testcases/PowerNVDump.py

```diff
@@ -305,6 +309,76 @@ def runTest(self):
         return boot_type
 
 
+class SBECrash_MPIPL(PowerNVDump):
+
+    def runTest(self):
+
+        if "OpenBMC" in self.bmc_type:
+            cmd = "mboxctl --reset"
+        elif "SMC" in self.bmc_type:
+            raise self.skipTest(
+                "SBE initiated MPIPL is not supported on SMC yet")
+        #run mboxctl reset on BMC
+        self.cv_BMC.run_command(cmd)
+        self.c.run_command("uname -a")
+        self.c.run_command("cat /etc/os-release")
+        #Get all chip ids on the machine
+        xscom_dirs = self.c.run_command(
+            "ls /sys/firmware/devicetree/base/ | grep xscom")
+        chip_ids = []
+        for node in xscom_dirs:
+            if node:
+                chip_id = self.c.run_command(
+                    "lsprop /sys/firmware/devicetree/base/%s/ibm,chip-id" % node)
+                chip_ids.append(chip_id[-1].strip())
+        chip_ids = sorted(chip_ids, reverse=True)
```

Comment:
Most of the time chip-id == `0` is primary chip. But we can't assume that.
Here you have to go through all xscom* node, look for primary property. If exists, then read chip-id property to get primary chip-id.

Reply:
Done